datafusion/sqllogictest/test_files/explain.slt

```diff
@@ -244,6 +244,159 @@ physical_plan DataSourceExec: file_groups={1 group: [[WORKSPACE_ROOT/datafusion/
 physical_plan_with_stats DataSourceExec: file_groups={1 group: [[WORKSPACE_ROOT/datafusion/core/tests/data/example.csv]]}, projection=[a, b, c], file_type=csv, has_header=true, statistics=[Rows=Absent, Bytes=Absent, [(Col[0]:),(Col[1]:),(Col[2]:)]]
 physical_plan_with_schema DataSourceExec: file_groups={1 group: [[WORKSPACE_ROOT/datafusion/core/tests/data/example.csv]]}, projection=[a, b, c], file_type=csv, has_header=true, schema=[a:Int32;N, b:Int32;N, c:Int32;N]
 
+# test SUM function with constant
+# Both the SUM(a + literal) and SUM(literal + a) should be optimized to SUM(a) + count(*) * literal
+query TT
+EXPLAIN VERBOSE SELECT SUM(a + 1), SUM(a + 2), SUM(3 + a) FROM simple_explain_test
```

Comment:
It's a bit unfortunate no other query has been affected 🤔 I think unfortunately it shows that this pattern will not be used a lot in the real world.

Reply:
I added some thoughts of generalizing this optimization here (which should benefit more common uses)
https://github.com/apache/datafusion/issues/15524#issuecomment-2771525677